File: python/grade-school/grade_school.py
class School:
    def __init__(self):
        self.student_roster = {}

    def add_student(self, name, grade):
        if grade not in self.student_roster:
            self.student_roster[grade] = [name]
        else:
            self.student_roster[grade].append(name)
        return "OK."

    def roster(self):
        roster = []
        for key in sorted(self.student_roster):
            roster = roster + sorted(self.student_roster[key])
        return roster

    def grade(self, grade_number):
        student_list = []
        if grade_number in self.student_roster:
            student_list = sorted(self.student_roster[grade_number])
        return student_list

File: ruby/grade-school/grade_school.rb
# frozen_string_literal: true

# Write your code for the 'Grade School' exercise in this file. Make the tests in
# `grade_school_test.rb` pass.
#
# To get started with TDD, see the `README.md` file in your
# `ruby/grade-school` directory.
class School
  attr_reader :classrooms

  def initialize
    @classrooms = {}
  end

  def students(grade)
    classrooms[grade].nil? ? [] : sort_students(grade)
  end

  def add(name, grade)
    if classrooms[grade].nil?
      classrooms[grade] = [name]
    else
      classrooms[grade].append(name)
    end
  end

  def students_by_grade
    return [] if classrooms.empty?

    students_list = []
    sort_students.each do |key, value|
      students_list.append({ grade: key, students: value.sort })
    end
    students_list
  end

  private

  def sort_students(grade = nil)
    return classrooms.sort if grade.nil?

    classrooms[grade].sort
  end
end
